W-15535654 Updates to improve SSO content jh (#32)
mulesoft/docs-api-experience-hub: modules/ROOT/pages/enabling-sso-for-your-portal.adoc

[[out-of-the-box-configurations]]
== Out-of-the-Box Configurations
API Experience Hub provides the following out-of-the-box configurations to assist the process of enabling SSO:

* anypoint_idp_id
+
The custom claim or SAML attribute provides the information that is received from the identity provider that determines which Anypoint Platform identity provider a user must be mapped to.

* AEH Users - ${salesforceOrganizationId}
+
The default Anypoint Experience Hub identity provider that's created in Anypoint Platform to map all self-registered users. This identity provider is used when the aeh_portal_idp is not set in the identity provider. Self-registered users are mapped with the default identity provider.

* AEH Portal - ${salesforceOrganizationId}_$*{salesforceCommunityId} 
+
This solution creates a team called AEH Portal in Access Management automatically. This team includes administrators, guests, and members profiles. You select this team when adding the group mapping. 

* AEHPortalRegistrationHandler.cls 
+
A single Apex class is provided in the API Experience Hub package called AEHPortalRegistrationHandler.cls that is compatible with SAML and OpenID Connect.  
+

> The registration handler is a piece of Apex code that runs when a user logs in to the Salesforce Experience Cloud site. You select this handler when configuring an authentication provider for Salesforce. When the identity provider is configured properly, Salesforce passes the portal ID to the AEHPortalRegistrationHandler. If the user does not exist, the registration handler creates the user's identity in Salesforce. If the user exists, the handler updates the user metadata based on information sent by the identity provider, for example, the user’s first name and last name. The registration handler must be contextualized with the portal ID in Salesforce to load specific configurations by the portal. For example, the user gets created in the profile and account related to the portal from the NetworkSelfRegistration object. 

Reviewer: ```suggestion
The registration handler is Apex code that runs when a user logs in to the Salesforce Experience Cloud site. You select this handler when configuring an authentication provider for Salesforce. When the identity provider is configured properly, Salesforce passes the portal ID to the AEHPortalRegistrationHandler. If the user does not exist, the registration handler creates the user's identity in Salesforce. If the user exists, the handler updates the user metadata based on information sent by the identity provider, for example, the user’s first name and last name. The registration handler uses the portal ID in Salesforce to load specific configurations by the portal. For example, the registration handler creates the user in the profile and account related to the portal from the NetworkSelfRegistration object. 
```

Reviewer: Is there any way to make some of these passive voices active? 

Does the handler get the portal information from the NetworkSelfRegistration object? Or does it use that object to create the user?

Reviewer: Is there a way to frontload with the user's task/value?

== Basic Terminology

The following terms are referenced in this topic:

> Authentication Provider::

Reviewer: Good use of definition list!

* Experience with identity management and setting up identity providers for SSO.
* Review Access Management's xref:access-management::external-identity.adoc[Identity Provider] documentation to understand identity management for Anypoint Platform.

* Be aware of what API Experience Hub expects for custom claims or attributes when creating a new application for the portal in the identity provider.
+

> Configuring the custom user attributes and claims for the identify provider ensures that the identity provider and the user information are mapped correctly. For example, the AEHPortalRegistrationHandler requires specific user information from the identity provider to create or update a user. The following required information lists the fields where this information is extracted from and the expected values. If there are more than one claim or attribute, the registration handler looks for the first expected value in the order it appears in the following tables:
> +

Reviewer 2: What do you think about making the following changes to this paragraph?
```suggestion
Configuring the user claims in the Identity Provider ensures that the user information is mapped correctly to a Salesforce's and Anypoint users. 
The AEHPortalRegistrationHandler requires specific user information from the Identity Provider to create or update a user. The following required information lists the information needed and which claims its going to be extracted from. If there are more than one claims for the same value, the registration handler will look the first expected claim in the order it appears in the following tables:
+
```

[%header%autowidth.spread]
|===

> |Field|Expected Values

Reviewer 2: I would change Expected Values for Expected Claims
```suggestion
|Field|Expected Claims
```

|First Name 
| given_name, first_name, firstname, and firstName
|Last Name 
| family_name, last_name, lastname, and lastName
|Email 
|email, email_address, and emailAddress
|Username 
|preferred_username, federation_identifier, email, sub, and NameId
|Groups | groups
|===
+ 

> The following information is optional except when using SAML to create an application for API Experience Hub in the identity provider:

Reviewer 2: ```suggestion
The following claims are optional and enable to configure specific use cases, for example when Configuring SSO to Use the Identity Provider That is Already Configured for Anypoint Platform.
```

* Experience with identity management and setting up identity providers for SSO.
* Review Access Management's xref:access-management::external-identity.adoc[Identity Provider] documentation to understand identity management for Anypoint Platform.

> * Be aware of what API Experience Hub expects for custom claims or attributes when creating a new application for the portal in the identity provider.

Reviewer 2: ```suggestion
* Be aware of what API Experience Hub expects for claims or attributes when creating a new application for the portal in the identity provider.
```

+

> Configuring the user claims in the identity provider ensures that the user information is mapped correctly to Salesforce  and Anypoint users.  

Reviewer 2: ```suggestion
Configuring the user claims in the identity provider ensures that the user information is mapped correctly to Salesforce and Anypoint users.  
```

+

> The AEHPortalRegistrationHandler requires specific user information from the identity provider to create or update a user. The following tables list the required information and which claims the information is extracted from. If there is more than one claim for the same value, the registration handler looks for the first expected claim in the order it appears in the following tables:

Reviewer: ```suggestion
The AEHPortalRegistrationHandler requires specific user information from the identity provider to create or update a user. The following tables list the required information and expected claims. If more than one claim exists for the same value, the registration handler uses the first expected claim in the order it appears in the following tables:
```

Reviewer: I think it would be helpful to explain what you mean by "expected claim" (the thing the handler extracts user information from?) earlier in this description. Then, you can just call it expected claim.

Reviewer: Should that be "following table" (or does this sentence also apply to the second table)? I ask because the second table doesn't mention claims.

Author: That is a good idea to explain what is meant by expected claims. I will add that.

+
[%header%autowidth.spread]
|===
|Field|Expected Claims
|First Name 
| given_name, first_name, firstname, and firstName
|Last Name 
| family_name, last_name, lastname, and lastName
|Email 
|email, email_address, and emailAddress
|Username 
|preferred_username, federation_identifier, email, sub, and NameId
|Groups | groups
|===
+ 

> The following information is optional and enables you to configure specific use cases like xref:sso-use-case-2.adoc[Configuring SSO When the Identity Provider Is Already Configured for Anypoint Platform]:

Reviewer: ```suggestion
To configure specific use cases, such as xref:sso-use-case-2.adoc[], use the following information:
```

+
[%header,cols="1a,2a,3a"]
|===
|Field |Description |Expected Values

> |Anypoint Idp Id|Used to map identities to the specific identity provider in Anypoint Platform. |anypoint_idp_id

Reviewer: ```suggestion
|Anypoint Idp Id|Map identities to the specific identity provider in Anypoint Platform. |anypoint_idp_id
```

> |Anypoint Username|This field is used to explicitly specify the Anypoint Platform user when the user's identity already exists in Anypoint Platform, but the username is different than the Salesforce user. |anypoint_username

Reviewer: ```suggestion
|Anypoint Username|Explicitly specify the Anypoint Platform user when the user's identity already exists in Anypoint Platform, but the username is different than the Salesforce user. |anypoint_username
```
